modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/api/extension/ExtensionModelDiscoverer.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.artifact.activation.api.extension;
+
+import org.mule.runtime.api.meta.model.ExtensionModel;
+import org.mule.runtime.module.artifact.activation.internal.extension.DefaultExtensionModelDiscoverer;
+import org.mule.runtime.module.artifact.activation.internal.extension.RepositoryLookupExtensionModelGenerator;
+import org.mule.runtime.module.artifact.api.classloader.ArtifactClassLoader;
+import org.mule.runtime.module.artifact.api.descriptor.ArtifactPluginDescriptor;
+
+import java.util.Set;
+import java.util.function.Function;
+
+/**
+ * Provides a way to discover {@link ExtensionModel}s from the Mule Runtime or from {@code mule-plugins} in the context of a
+ * deployable artifact.
+ * 
+ * @since 4.5
+ */
+public interface ExtensionModelDiscoverer {
+
+  /**
+   * Creates an {@link ExtensionModelDiscoverer} that will rely on class introspection for generating the extension models for
+   * plugins developed with the Java SDK.
```

Comment:
not necessarily developed with the Java SDK. The contract should be returning the models in a classloader.